### ISSUE TITLE
Typescript definitions broken for react-native

### ISSUE THREAD

**Issue by iRoachie** [rt2zz/redux-persist#395]
Using persistStore with react native and AsyncStorage throws an error.

file: 'file:///Users/kyleroach/Desktop/wwwpromoter-mobile/src/store.ts'
severity: 'Error'
message: 'Argument of type '{ storage: AsyncStorageStatic; whitelist: string[]; }' is not assignable to parameter of type 'PersistorConfig'.
  Types of property 'storage' are incompatible.
    Type 'AsyncStorageStatic' is not assignable to type 'Storage'.
      Types of property 'getItem' are incompatible.
        Type '(key: string, callback?: (error?: Error, result?: string) => void) => Promise<string>' is not assignable to type '<Result>(key: string, onComplete?: OnComplete<Result>) => Promise<Result>'.
          Types of parameters 'callback' and 'onComplete' are incompatible.
            Types of parameters 'result' and 'result' are incompatible.
              Type 'string' is not assignable to type 'Result'.'
at: '15,21'
source: 'ts'

**Comment by jaycanty** [comment]
Yes, I am getting this error too with `persistStore(store, {storage: AsyncStorage})`

I've tried to wrap AsyncStore but cannot figure out to return a string type (returned from AsyncStore.getItem callback as the generic <Result> type required in Storage interface.

```
    public getItem<Result>(key: string, onComplete?: OnComplete<Result>): Promise<Result> {
        const callBack = (error?: Error, result?: string) => {
            onComplete(error, result as Result)  // <= Cannot cast string to Result
        }
```

Maybe this explains the gist.  I am new to this whole scene....

**Comment by sdiaz** [comment]
👍

**Comment by iRoachie** [comment]
Sent PR to fix #399

**Comment by iRoachie** [comment]
Fixed by #399
